Trapeze.cs

```diff
@@ -0,0 +1,84 @@
+
+class Trapeze : Shape
+{
+    int TopBase;
+    int Height;
+    public Trapeze(bool FillShape, char Char, int TopBase, int Height) : base(FillShape, Char)
+    {
+        this.TopBase = TopBase;
+        this.Height = Height;
+    }
+
+    public override void PrintShape()
+    {
+        for (int i=0; i<this.Height; i++)
+        {
+            for (int j = 0; j <= 2*(this.Height-1)+this.TopBase; j++)
+            {
+                if (!this.FillShape) // hollow
+                {
+                    if ((i == 0 && j >= this.Height-1 && j <= this.TopBase + this.Height-1) // top base
+                        || (i == this.Height-1) // bottom base
+                        || (j == this.Height-1-i) // left wing
+                        || (j == this.Height-1 + this.TopBase + i)) // right wing
```

Comment:
Imagine having such a function- cleaner code and well-understood.
```suggestion
                    if (IsFoo(i, j))
```